addons/content-index/src/main/java/org/commonjava/indy/content/index/IndexingContentManagerDecorator.java

```diff
@@ -360,22 +363,25 @@ else if ( indexCfg.isAuthoritativeIndex() && store.isAuthoritativeIndex() )
             if ( !nfc.isMissing( resource ) )
             {
                 logger.debug( "No group index hits. Devolving to member store indexes." );
-                transfer = getIndexedMemberTransfer( (Group) store, path, op, (member)->{
+                for (StoreKey memberKey: ((Group)store).getConstituents()){
                     try
                     {
-                        return delegate.getTransfer( member, path, op );
+                        ArtifactStore member = storeDataManager.getArtifactStore( memberKey );
+                        if ( member != null )
+                        {
+                            transfer = getTransfer( member, path, op );
+                        }
                     }
-                    catch ( IndyWorkflowException e )
+                    catch ( IndyDataException e )
                     {
-                        logger.error( String.format(
-                                "Failed to getTransfer() for: %s:%s with operation: %s. Reason: %s",
-                                member.getKey(), path, op, e.getMessage() ), e );
+                        logger.error( String.format( "Failed to lookup store: %s (in membership of: %s). Reason: %s",
+                                                     memberKey, store.getKey(), e.getMessage() ), e );
                     }
-
-                    return null;
-                } );
+                }
                 if ( transfer != null )
                 {
+                    logger.debug( "Indexing transfer: {}", transfer );
```

Comment:
Do we need to do existence check here as well as a null check? I can't remember if that's handled at a lower level.